### PR TITLE
Remove heartbeats completely

### PR DESCRIPTION
**Pull request by levitte**
##### Checklist

- [x] CLA is signed

### CODE REVIEW

**Comment by levitte** [comment]
Re-awakening an old PR

**Review by t-j-h — APPROVED** [review]

**Comment by richsalz** [comment]
This should probably have a CHANGES or NEWS entry.

**Comment by levitte** [comment]
Entries added.  Also, it turns out that this fixes #4856, so I added a reference to that issue as well in the squash commit

**Review by t-j-h — APPROVED** [review]

**Comment by levitte** [comment]
Merged.

558ea84743918f7a93bfbfc259f86ad1fa4c8de9 Remove heartbeats completely

**Comment by mattcaswell** [comment]
Why wasn't the configure option moved to the list of deprecated Configure options. Now if someone explicitly compiles with "no-heartbeat" they will get an error rather than it being ignored - which seems to introduce unnecessary breakage.

File: ssl/t1_trce.c [hunk 1]
```diff
@@ -468,7 +468,6 @@ static const ssl_trace_tbl ssl_exts_tbl[] = {
     {TLSEXT_TYPE_srp, "srp"},
     {TLSEXT_TYPE_signature_algorithms, "signature_algorithms"},
     {TLSEXT_TYPE_use_srtp, "use_srtp"},
-    {TLSEXT_TYPE_heartbeat, "tls_heartbeat"},
```
Comment (FdaSilvaYY) [inline ssl/t1_trce.c]: Was it really necessary to remove this trace code ?

Comment (levitte) [reply ssl/t1_trce.c]: If there's no heartbeat code, what good does that trace do you?

**Pull request closed without merging**